In [120]:
# Standard Data Science Helpers
{
    "tags": [
        "remove-cell"
    ]
}

import numpy as np
import pandas as pd
import scipy

import plotly.graph_objs as go
import plotly.graph_objects as go
import plotly.express as px

from ipywidgets import widgets

import chart_studio.plotly as py

from plotly.offline import iplot, init_notebook_mode
init_notebook_mode(connected=True)

import cufflinks as cf
cf.go_offline(connected=True)
cf.set_config_file(colorscale='plotly', world_readable=True)

# Extra options
pd.options.display.max_rows = 30
pd.options.display.max_columns = 25

# Show all code cells outputs
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = 'all'

import os
from IPython.display import Image, display, HTML

import ipywidgets as widgets
from ipywidgets import interact, interact_manual

{'tags': ['remove-cell']}

In [121]:
{
    "tags": [
        "remove-cell"
    ]
}
nt_df = pd.read_csv('nt_crime_statistics_may_2022.csv')

{'tags': ['remove-cell']}

In [122]:
{
    "tags": [
        "remove-cell"
    ]
}
nt_df.keys()

{'tags': ['remove-cell']}

Index(['As At', 'Year', 'Month number', 'Offence category', 'Offence type',
       'Alcohol involvement', 'DV involvement', 'Reporting region ',
       'Statistical Area 2', 'Number of offences'],
      dtype='object')

In [123]:
{
    "tags": [
        "remove-input"
    ]
}
nt_df.columns = [colname.strip() for colname in nt_df.columns]
nt_df.drop('As At', axis = 1)

{'tags': ['remove-input']}

Year  Month number                                   Offence category  \
0      2008             1  Abduction - harassment and other offences agai...   
1      2008             1  Abduction - harassment and other offences agai...   
2      2008             1  Abduction - harassment and other offences agai...   
3      2008             1  Abduction - harassment and other offences agai...   
4      2008             1  Abduction - harassment and other offences agai...   
...     ...           ...                                                ...   
44989  2022             5         Theft and related offences (other than MV)   
44990  2022             5         Theft and related offences (other than MV)   
44991  2022             5         Theft and related offences (other than MV)   
44992  2022             5         Theft and related offences (other than MV)   
44993  2022             5         Theft and related offences (other than MV)   

                                     Offence type Alcohol involvement  \
0      Deprivation of liberty /false imprisonment                   -   
1      Deprivation of liberty /false imprisonment                   -   
2      Deprivation of liberty /false imprisonment                   -   
3                 Harassment and private nuisance                   -   
4                 Harassment and private nuisance                   -   
...                                           ...                 ...   
44989  Theft and related offences (other than MV)                   -   
44990  Theft and related offences (other than MV)                   -   
44991  Theft and related offences (other than MV)                   -   
44992  Theft and related offences (other than MV)                   -   
44993  Theft and related offences (other than MV)                   -   

      DV involvement Reporting region    Statistical Area 2  \
0                  -    Alice Springs                     -   
1                  -       NT Balance             Alligator   
2                  -    Tennant Creek                     -   
3                  -       NT Balance             Alligator   
4                  -       NT Balance                  Daly   
...              ...              ...                   ...   
44989              -       NT Balance               Weddell   
44990              -       NT Balance           West Arnhem   
44991              -       NT Balance  Yuendumu - Anmatjere   
44992              -       Palmerston                     -   
44993              -    Tennant Creek                     -   

       Number of offences  
0                       1  
1                       1  
2                       1  
3                       2  
4                       1  
...                   ...  
44989                   5  
44990                   8  
44991                   8  
44992                  70  
44993                   7  

[44994 rows x 9 columns]

In [124]:
{
    "tags": [
        "remove-cell"
    ]
}
nt_df.shape

{'tags': ['remove-cell']}

(44994, 10)

In [125]:
{
    "tags": [
        "remove-cell"
    ]
}
nt_df.keys()

{'tags': ['remove-cell']}

Index(['As At', 'Year', 'Month number', 'Offence category', 'Offence type',
       'Alcohol involvement', 'DV involvement', 'Reporting region',
       'Statistical Area 2', 'Number of offences'],
      dtype='object')

In [126]:
{
    "tags": [
        "remove-cell"
    ]
}
region_year_offences_df = pd.DataFrame(nt_df.groupby(['Reporting region','Year'])['Number of offences'].sum().reset_index())
region_year_offences_df

{'tags': ['remove-cell']}

Reporting region  Year  Number of offences
0      Alice Springs  2008                5312
1      Alice Springs  2009                5735
2      Alice Springs  2010                6989
3      Alice Springs  2011                6235
4      Alice Springs  2012                7418
..               ...   ...                 ...
100    Tennant Creek  2018                1107
101    Tennant Creek  2019                1344
102    Tennant Creek  2020                1277
103    Tennant Creek  2021                1451
104    Tennant Creek  2022                 470

[105 rows x 3 columns]

In [127]:
{
    "tags": [
        "remove-cell"
    ]
}
region_month_year_df = pd.DataFrame(nt_df.groupby(['Reporting region', 'Year', 'Month number'])['Number of offences'].sum().reset_index())
region_month_year_df

{'tags': ['remove-cell']}

Reporting region  Year  Month number  Number of offences
0       Alice Springs  2008             1                 420
1       Alice Springs  2008             2                 358
2       Alice Springs  2008             3                 508
3       Alice Springs  2008             4                 487
4       Alice Springs  2008             5                 406
...               ...   ...           ...                 ...
1206    Tennant Creek  2022             1                  63
1207    Tennant Creek  2022             2                  79
1208    Tennant Creek  2022             3                 126
1209    Tennant Creek  2022             4                 119
1210    Tennant Creek  2022             5                  83

[1211 rows x 4 columns]

In [128]:
{
    "tags": [
        "remove-input"
    ]
}
region_month_year_offence_category = pd.DataFrame(nt_df.groupby(['Reporting region', 'Year', 'Month number', 'Offence category', 'Offence type'])['Number of offences'].sum().reset_index())
region_month_year_offence_category

{'tags': ['remove-input']}

Reporting region  Year  Month number  \
0        Alice Springs  2008             1   
1        Alice Springs  2008             1   
2        Alice Springs  2008             1   
3        Alice Springs  2008             1   
4        Alice Springs  2008             1   
...                ...   ...           ...   
14752    Tennant Creek  2022             5   
14753    Tennant Creek  2022             5   
14754    Tennant Creek  2022             5   
14755    Tennant Creek  2022             5   
14756    Tennant Creek  2022             5   

                                        Offence category  \
0      Abduction - harassment and other offences agai...   
1      Abduction - harassment and other offences agai...   
2                          Acts intended to cause injury   
3                                   Commercial break-ins   
4                                   Commercial break-ins   
...                                                  ...   
14752                   Motor theft and related offences   
14753                   Motor theft and related offences   
14754  Other Dangerous or negligent acts endangering ...   
14755                           Property Damage Offences   
14756         Theft and related offences (other than MV)   

                                            Offence type  Number of offences  
0             Deprivation of liberty /false imprisonment                   1  
1                                  Threatening behaviour                   4  
2                                                Assault                  95  
3                            Commercial break-ins actual                  31  
4                         Commercial break-ins attempted                   1  
...                                                  ...                 ...  
14752                     Illegal use of a motor vehicle                   1  
14753           Theft of motor vehicle parts or contents                   2  
14754  Other dangerous or negligent acts endangering ...                   2  
14755                           Property damage offences                  35  
14756         Theft and related offences (other than MV)                   7  

[14757 rows x 6 columns]

In [129]:
{
    "tags": [
        "remove-cell"
    ]
}
population = []
for region in region_year_offences_df['Reporting region']:
    if(region == 'Alice Springs'):
        population.append(23736)
    elif(region == 'Darwin'):
        population.append(132045)
    elif(region == 'Katherine'):
        population.append(9717)
    elif(region == 'NT Balance'):
        population.append(4576)    
    elif(region == 'Nhulunbuy'):
        population.append(3240)
    elif(region == 'Palmerston'):
        population.append(29346)
    elif(region == 'Tennant Creek'):
        population.append(2991)

{'tags': ['remove-cell']}

In [130]:
{
    "tags": [
        "remove-cell"
    ]
}
unemployed = []
for region in region_year_offences_df['Reporting region']:
    if(region == 'Alice Springs'):
        unemployed.append(430)
    elif(region == 'Darwin'):
        unemployed.append(3466)
    elif(region == 'Katherine'):
        unemployed.append(162)
    elif(region == 'NT Balance'):
        unemployed.append(4153)
    elif(region == 'Nhulunbuy'):
        unemployed.append(52)
    elif(region == 'Palmerston'):
        unemployed.append(922)
    elif(region == 'Tennant Creek'):
        unemployed.append(89)

{'tags': ['remove-cell']}

In [14]:
pop_unemp = pd.DataFrame({'population': population, 'unemployed': unemployed})
pop_unemp

In [131]:
{
    "tags": [
        "remove-input"
    ]
}
region_population_unemployment_df = pd.concat([region_year_offences_df, pop_unemp], axis = 1)
region_population_unemployment_df

{'tags': ['remove-input']}

Reporting region  Year  Number of offences  population  unemployed
0      Alice Springs  2008                5312       23736         430
1      Alice Springs  2009                5735       23736         430
2      Alice Springs  2010                6989       23736         430
3      Alice Springs  2011                6235       23736         430
4      Alice Springs  2012                7418       23736         430
..               ...   ...                 ...         ...         ...
100    Tennant Creek  2018                1107        2991          89
101    Tennant Creek  2019                1344        2991          89
102    Tennant Creek  2020                1277        2991          89
103    Tennant Creek  2021                1451        2991          89
104    Tennant Creek  2022                 470        2991          89

[105 rows x 5 columns]

In [132]:
{
    "tags": [
        "remove-input"
    ]
}
offence_category_type = pd.DataFrame(nt_df.groupby(['Offence category','Offence type'])['Number of offences'].sum()).reset_index()
offence_category_type

{'tags': ['remove-input']}

Offence category  \
0   Abduction - harassment and other offences agai...   
1   Abduction - harassment and other offences agai...   
2   Abduction - harassment and other offences agai...   
3   Abduction - harassment and other offences agai...   
4                       Acts intended to cause injury   
5                       Acts intended to cause injury   
6                                Commercial break-ins   
7                                Commercial break-ins   
8                       Homicide and related Offences   
9                       Homicide and related Offences   
10                      Homicide and related Offences   
11                      Homicide and related Offences   
12                                    House break-ins   
13                                    House break-ins   
14                   Motor theft and related offences   
15                   Motor theft and related offences   
16  Other Dangerous or negligent acts endangering ...   
17                           Property Damage Offences   
18           Robbery - extortion and related offences   
19           Robbery - extortion and related offences   
20                Sexual Assault and related offences   
21                Sexual Assault and related offences   
22         Theft and related offences (other than MV)   

                                         Offence type  Number of offences  
0                            Abduction and kidnapping                  15  
1          Deprivation of liberty /false imprisonment                 610  
2                     Harassment and private nuisance                1088  
3                               Threatening behaviour                3314  
4                                             Assault              102368  
5                 Other acts intended to cause injury                 371  
6                         Commercial break-ins actual               26171  
7                      Commercial break-ins attempted                3429  
8                                    Attempted Murder                  45  
9                               Driving causing death                  37  
10                                       Manslaughter                  45  
11                                             Murder                 121  
12                             House break-ins actual               25953  
13                          House break-ins attempted                3543  
14                     Illegal use of a motor vehicle               13289  
15           Theft of motor vehicle parts or contents               16197  
16  Other dangerous or negligent acts endangering ...                1787  
17                           Property damage offences              108292  
18                            Blackmail and extortion                  28  
19                                            Robbery                1781  
20                     Non-assaultive sexual offences                 796  
21                                     Sexual assault                5452  
22         Theft and related offences (other than MV)              105809

In [133]:
{
    "tags": [
        "remove-input"
    ]
}
offence_category = pd.DataFrame(nt_df.groupby('Offence category')['Number of offences'].sum()).reset_index()
offence_category

{'tags': ['remove-input']}

Offence category  Number of offences
0   Abduction - harassment and other offences agai...                5027
1                       Acts intended to cause injury              102739
2                                Commercial break-ins               29600
3                       Homicide and related Offences                 248
4                                     House break-ins               29496
5                    Motor theft and related offences               29486
6   Other Dangerous or negligent acts endangering ...                1787
7                            Property Damage Offences              108292
8            Robbery - extortion and related offences                1809
9                 Sexual Assault and related offences                6248
10         Theft and related offences (other than MV)              105809

[Current - Northern Territory Crime Statistics May 2022](https://data.nt.gov.au/dataset/current-northern-territory-crime-statistics-may-2022)

[South Australian Government Data Directory](https://data.sa.gov.au/data/dataset/crime-statistics)

[Australian Bureau of Statistics](https://www.abs.gov.au/census/find-census-data/search-by-area)

"Foreign travellers warned about visiting Alice Springs". The New Daily. 15 March 2017. Retrieved 9 October 2019.

Webb, Carolyn (15 March 2017). "Foreign tourists warned on Alice Springs safety risks". The Sydney Morning Herald. Retrieved 9 October 2019.

["The Northern Territory is the Murder Capital of Australia."] (https://www.theguardian.com/news/datablog/2013/jun/18/australia-new-crime-statistics) The Guardian. 18 June 2013. Retrieved 15 August 2021 



![Murder Capital of Australia](murder_capital.png)

Australia's Northern Territory has acquired quite a reputation due to *its consistently higher per capita rate for homicide and manslaughter than other Australian states and territories, as well as a high level of assaults.* ([Wikipedia](https://en.wikipedia.org/wiki/Crime_in_the_Northern_Territory)) Most of these were related to domestic violence and alcohol intake. 

The headline presented above came from an article which was written for the [Guardian](https://www.theguardian.com/news/datablog/2013/jun/18/australia-new-crime-statistics) by Nick Evershed on the 18th of June in 2013. He wrote that the Northern Territory's murder and assault rates are the highest in the country. 

The aim of this exploratory data analysis is to look into the crime trends in Australia's Northern territory using the Pandas and Plotly. The dataset used was the most recently updated [Current - Northern Territory Crime Statistics May 2022](https://data.nt.gov.au/dataset/current-northern-territory-crime-statistics-may-2022) from the Northern Territory Government website.

<h2 style = 'color: blue'> Reported Offences in the Northern Territory </h2>

In [134]:
{
    "tags": [
        "remove-intput"
    ]
}

@interact
def show_category_trend_by_region (By_Region = True,
                                   Region = list(region_month_year_offence_category['Reporting region'].unique()), 
                                   By_Year = False,
                                   Year = list(region_month_year_offence_category['Year'].unique()),
                                   By_Month = False,
                                   Month_Number = list(region_month_year_offence_category['Month number'].unique())):
    
    region_filter = region_month_year_offence_category['Reporting region'] == Region
    year_filter = region_month_year_offence_category['Year'] == Year
    month_filter = region_month_year_offence_category['Month number'] == Month_Number
    
    if By_Region:
        if By_Month & By_Year:
            region_month_year_offence_category[region_filter & year_filter & month_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence category', title=f'Reported Offences in Month {Month_Number} of {Year} in {Region}')
        elif By_Year and not By_Month:
            region_month_year_offence_category[region_filter & year_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence category', title=f'Reported Offences in {Year} in {Region}')    
        elif By_Month and not By_Year:
            region_month_year_offence_category[region_filter & month_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence category', title=f'Reported Offences in Month {Month_Number} from 2008 to 2022 in {Region}')
        else:
            region_month_year_offence_category[region_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence category', title=f'Reported Offences from 2008 to 2022 in {Region}')
    else:
        if By_Month & By_Year:
            region_month_year_offence_category[year_filter & month_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence category', title=f'Reported Offences in Month {Month_Number} of {Year}')
        elif By_Year and not By_Month:
            region_month_year_offence_category[year_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence category', title=f'Reported Offences in {Year}')    
        elif By_Month and not By_Year:
            region_month_year_offence_category[month_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence category', title=f'Reported Offences in Month {Month_Number} from 2008 to 2022')
        else:
            region_month_year_offence_category.iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence category', title=f'Reported Offences from 2008 to 2022')


{'tags': ['remove-intput']}

interactive(children=(Checkbox(value=True, description='By_Region'), Dropdown(description='Region', options=('…

In [135]:
{
    "tags": [
        "remove-intput"
    ]
}

@interact
def show_offence_type(Category = list(region_month_year_offence_category['Offence category'].unique()),
                      By_Year = False,
                      Year = list(region_month_year_offence_category['Year'].unique()),
                      By_Month = False,
                      Month_Number = list(region_month_year_offence_category['Month number'].unique())):
    
    category_filter = region_month_year_offence_category['Offence category'] == Category
    year_filter = region_month_year_offence_category['Year'] == Year
    month_filter = region_month_year_offence_category['Month number'] == Month_Number
    
    
    if By_Month & By_Year:
        region_month_year_offence_category[category_filter & year_filter & month_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence type', title=f'Types of Reported {Category} in Month {Month_Number} of {Year}')
    elif By_Year and not By_Month:
        region_month_year_offence_category[category_filter & year_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence type', title=f'Types of Reported {Category} in {Year}')    
    elif By_Month and not By_Year:
        region_month_year_offence_category[category_filter & month_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence type', title=f'Types of Reported {Category} in Month {Month_Number} from 2008 to 2022')
    else:
        region_month_year_offence_category[category_filter].iplot(kind = 'pie', values = 'Number of offences', labels = 'Offence type', title=f'Types of Reported {Category} from 2008 to 2022')


{'tags': ['remove-intput']}

interactive(children=(Dropdown(description='Category', options=('Abduction - harassment and other offences aga…

> ### The top 3 reported offences in the Northern Territory from 2008 to 2022 are: 
>
> - Property Damage Offences: 25.8%
> - Theft and Related Offences (other than Motor Vehicle Theft): 25.2%
> - Acts intended to cause injury: 24.4%
>

Homicide and related offences only make up 0.059% of the reported offences. With a total of 44,994 reported offences, this means that 248 incidences of homicide occurred in a span of 13.5 years, which can indicate that an average of 18 homicides are reported each year. 

> ### The offence categories are subdivided into different types.
>
> - Looking at the different types of offences associated with homicide, it is interesting to see that from 2008 to 2022, only 48.8% was classified as murder, which means that only half of the homicide offences, were intentional. 

 - 18.1% of the homicide offences fall under manslaughter, which means that it was not intentional. This mean that around 45 out of 248 incidents could be related to violence due to alcohol or negligent behaviour. 

 - Considering that acts intended to cause injury and dangerous, negligent acts that endanger persons, abduction, and sexual assault make up more than a quarter of the reported offences, it is highly possible for these acts to lead to manslaughter.

 - Theft and break-ins, which make up half of the reported incidents, can also lead to manslaughter should struggles between the victim and thief occur. 

 - 14.9% of the homicide cases were due to driving accidents (which can easily be caused by alcohol and negligent behaviour.)

 - Lastly, 18.1% of the cases reported were classified as attempted murder, which should mean that the victim did not die from it. 

In [136]:
{
    "tags": [
        "remove-intput"
    ]
}
nt_df['Number of offences'].describe()

{'tags': ['remove-intput']}

count    44994.000000
mean         9.346602
std         26.984379
min          1.000000
25%          1.000000
50%          3.000000
75%          6.000000
max        458.000000
Name: Number of offences, dtype: float64

In [137]:
{
    "tags": [
        "remove-intput"
    ]
}
nt_df[nt_df['Number of offences'] == 458]

{'tags': ['remove-intput']}

As At  Year  Month number          Offence category  \
6111  1/07/2022  2010             1  Property Damage Offences   

                  Offence type Alcohol involvement DV involvement  \
6111  Property damage offences                   -              -   

     Reporting region Statistical Area 2  Number of offences  
6111           Darwin                  -                 458

In [138]:
{
    "tags": [
        "remove-intput"
    ]
}
@interact
def show_max_min_offence(Category = list(region_month_year_offence_category['Offence category'].unique())):
    print(region_month_year_offence_category[region_month_year_offence_category['Offence category'] == Category]['Number of offences'].describe())

{'tags': ['remove-intput']}

interactive(children=(Dropdown(description='Category', options=('Abduction - harassment and other offences aga…

<h2 style = 'color: blue'> Regional Trend </h2>

> *In 2017, the governments of the UK, Germany and Canada issued travel advisories to their citizens travelling in Australia, singling out Alice Springs as a place where extra precautions should be taken: "Take care in the town centre of Alice Springs at night. There have been a number of incidents of harassment, robberies and attacks (including sexual assault) on foreign tourists." (UK).*
>
> -- "Foreign travellers warned about visiting Alice Springs". The New Daily. 15 March 2017. Retrieved 9 October 2019.


In [139]:
{
    "tags": [
        "remove-intput"
    ]
}
px.line(region_year_offences_df, x = 'Year', y = 'Number of offences', color = 'Reporting region', markers = True)

{'tags': ['remove-intput']}

In [140]:
{
    "tags": [
        "remove-intput"
    ]
}

@interact
def show_offences_percentage_reporting_region(Category = list(region_month_year_offence_category['Offence category'].unique())):
    region_month_year_offence_category[region_month_year_offence_category['Offence category'] == Category].iplot(kind = 'pie', 
                                                                                                               values = 'Number of offences',
                                                                                                                labels = 'Reporting region',
                                                                                                                title = f'{Category} in NT Regions From 2008 to 2022')

{'tags': ['remove-intput']}

interactive(children=(Dropdown(description='Category', options=('Abduction - harassment and other offences aga…

In [141]:
{
    "tags": [
        "remove-intput"
    ]
}

@interact
def show_offences_per_reporting_region(Region = list(region_month_year_offence_category['Reporting region'].unique())):
    region_month_year_offence_category[region_month_year_offence_category['Reporting region'] == Region].groupby('Offence category')['Number of offences'].sum().iplot(kind = 'barh',
                                                                                                             xTitle = 'Month', 
                                                                                                             yTitle = 'Number of Offences',
                                                                                                             title = f'Number of Offences in {Region} from 2008 - 2022')

{'tags': ['remove-intput']}

interactive(children=(Dropdown(description='Region', options=('Alice Springs', 'Darwin', 'Katherine', 'NT Bala…

In [142]:
{
    "tags": [
        "remove-input"
    ]
}

px.scatter(region_population_unemployment_df, x = 'Year', y = 'Number of offences', size = 'population', color = 'Reporting region', title = 'Population vs Unemployment vs Number of Offenses' )

{'tags': ['remove-intput']}

In [151]:
{
    "tags": [
        "remove-input"
    ]
}

px.scatter(region_population_unemployment_df, x = 'Year', y = 'Number of offences', size = 'unemployed', color = 'Reporting region', title = 'Unemployment vs Number of Offenses' )

{'tags': ['remove-input']}

<h2 style = 'color: blue'> Yearly Trend </h2>

According to the source of the data, [Current - Northern Territory Crime Statistics May 2022](https://data.nt.gov.au/dataset/current-northern-territory-crime-statistics-may-2022), the *offences in the Northern Territory can show **strong seasonal patterns, with predictable peaks and troughs at certain times of the year.**  Therefore, comparisons should be made of the same months in different years, rather than consecutive months.  For trends, a 12-month rolling average or rolling sum is recommended in order to dampen the month-to-month variability and account for the seasonal cycles.*

In [150]:
{
    "tags": [
        "remove-input"
    ]
}

@interact
def show_yearly_crime_count(Year=(2008, 2022, 1)):
    
    monthly_sum = nt_df[nt_df['Year']==Year].groupby('Month number')['Number of offences'].sum()
    yearly_mean = nt_df[nt_df['Year']==Year].groupby('Month number')['Number of offences'].sum().mean()
    monthly_variance = monthly_sum - yearly_mean
    maximum_sum = nt_df[nt_df['Year']==Year].groupby('Month number')['Number of offences'].sum().max()
    minimum_sum = nt_df[nt_df['Year']==Year].groupby('Month number')['Number of offences'].sum().min()
    years = nt_df['Year'].unique
    
    yearly_count = pd.DataFrame(monthly_sum)
    yearly_count['Mean'] = yearly_mean
    yearly_count['Variance'] = monthly_variance 
    yearly_count['Percent Variance'] = monthly_variance / yearly_mean * 100
    
    yearly_max_min = pd.DataFrame([maximum_sum, minimum_sum])
    
    if len(yearly_count.index) > 5:
        yearly_count.index = ['Jan', 'Feb', 'March', 'April', 'May', 'June', 'July', 'Aug', 'Sept', 'Oct', 'Nov', 'Dec']
    else:
        yearly_count.index = ['Jan', 'Feb', 'March', 'April', 'May']
                 

    yearly_count.iplot(kind='bar',
                       barmode='group', 
                       title=f'Number of Offences, Mean & Variance For Each Month in {Year}'
                      )
    
    display(yearly_count)
    

{'tags': ['remove-input']}

interactive(children=(IntSlider(value=2015, description='Year', max=2022, min=2008), Output()), _dom_classes=(…

In [149]:
{
    "tags": [
        "remove-input"
    ]
}

@interact
def show_monthly_crime_count(Month=[('January', 1), ('February', 2), ('March', 3), ('April', 4), ('May', 5), ('June', 6), ('July', 7), ('August', 8), ('September', 9), ('October', 10), ('November', 11), ('December', 12)]):
    

    monthly_sum_per_year = nt_df[nt_df['Month number'] == Month].groupby('Year')['Number of offences'].sum()
    mean_by_month = nt_df[nt_df['Month number'] == Month].groupby('Year')['Number of offences'].sum().mean()
    variance_by_month = monthly_sum_per_year - mean_by_month
    
    monthly_count = pd.DataFrame(monthly_sum_per_year)
    monthly_count['Mean'] = mean_by_month
    monthly_count['Variance'] = variance_by_month 
    monthly_count['Percent Variance'] = variance_by_month / mean_by_month * 100

    monthly_count.iplot(kind='bar',
                       barmode='group', 
                       title=f'Number of Offences, Mean & Variance for Each Month Grouped By Year'
                      )
    
    display(monthly_count)



{'tags': ['remove-input']}

interactive(children=(Dropdown(description='Month', options=(('January', 1), ('February', 2), ('March', 3), ('…

In [147]:
{
    "tags": [
        "remove-cell"
    ]
}
years = nt_df['Year'].unique()
months = nt_df['Month number'].unique()
offence_sum = []

for month in months: 
    offence_sum.append(nt_df[nt_df['Month number'] == month].groupby('Year')['Number of offences'].sum())

Offence_sum_df = pd.DataFrame(offence_sum)
Offence_sum_df = Offence_sum_df.set_index(months)
Offence_sum_df[2022] = Offence_sum_df[2022].fillna(0)

{'tags': ['remove-cell']}

In [35]:
Offence_sum_df = Offence_sum_df.transpose()

In [36]:
Offence_sum_df = Offence_sum_df.stack().reset_index()

In [37]:
Offence_sum_df = Offence_sum_df.rename(columns={'level_1': 'Month', 0: 'Number of Offences'})

In [148]:
{
    "tags": [
        "remove-input"
    ]
}
Offence_sum_df 

{'tags': ['remove-input']}

Year    2008    2009    2010    2011    2012    2013    2014    2015    2016  \
1     2420.0  2669.0  3072.0  2533.0  2880.0  2488.0  2305.0  2776.0  2503.0   
2     2417.0  2183.0  2565.0  2494.0  2578.0  2301.0  1963.0  2103.0  2337.0   
3     2630.0  2451.0  2809.0  2367.0  2788.0  2555.0  2451.0  2466.0  2340.0   
4     2583.0  2345.0  2611.0  2228.0  2862.0  2339.0  2429.0  2321.0  2287.0   
5     2262.0  2268.0  2654.0  2095.0  2526.0  2276.0  2328.0  2130.0  2336.0   
6     2276.0  2217.0  2279.0  1990.0  2087.0  2224.0  2364.0  2310.0  2284.0   
7     2204.0  2174.0  2536.0  2195.0  2360.0  2375.0  2352.0  2042.0  2271.0   
8     2198.0  2144.0  2212.0  1991.0  1954.0  2022.0  2196.0  2114.0  2178.0   
9     2228.0  2410.0  2489.0  2078.0  2198.0  1922.0  2093.0  2131.0  2256.0   
10    2498.0  2661.0  2624.0  2186.0  2465.0  1980.0  2221.0  2237.0  2439.0   
11    2609.0  2467.0  2311.0  2430.0  2449.0  2182.0  2288.0  2277.0  2469.0   
12    2342.0  2604.0  2376.0  2695.0  2491.0  2142.0  2674.0  2433.0  2706.0   

Year    2017    2018    2019    2020    2021    2022  
1     3103.0  2906.0  2874.0  2924.0  2901.0  3370.0  
2     2622.0  2637.0  2220.0  2948.0  2523.0  2698.0  
3     2560.0  2588.0  2615.0  2991.0  2993.0  3359.0  
4     2225.0  2718.0  2324.0  1782.0  2713.0  3299.0  
5     2534.0  2556.0  2464.0  1571.0  2680.0  2728.0  
6     2160.0  2146.0  1991.0  1632.0  2284.0     0.0  
7     2218.0  2481.0  2063.0  1688.0  2377.0     0.0  
8     2394.0  2428.0  2266.0  1894.0  2779.0     0.0  
9     2236.0  2511.0  2154.0  2297.0  2852.0     0.0  
10    2533.0  2529.0  2645.0  2820.0  2983.0     0.0  
11    2419.0  2478.0  2606.0  2816.0  2923.0     0.0  
12    2886.0  2570.0  2627.0  2724.0  2771.0     0.0

In [39]:
fig1 = px.violin(Offence_sum_df[Offence_sum_df['Year'] != 2022], 
                x="Month", 
                y="Number of Offences", 
                color='Month', 
                box=True, 
                points='all',
                title = 'A Comparison of Reported Offences By Month From 2008 to 2021')

fig1.show()


In [40]:
Offence_sum_df_no_2020 = Offence_sum_df[(Offence_sum_df['Year'] != 2020) & (Offence_sum_df['Year'] != 2022)] 
fig1 = px.violin(Offence_sum_df_no_2020, 
                x="Month", 
                y="Number of Offences", 
                color='Month', 
                box=True, 
                points='all',
                title = 'A Comparison of Reported Offences By Month From 2008 to 2019, 2021 ')

fig1.show()

In [41]:
year_month_df = pd.DataFrame(nt_df.groupby(['Year', 'Month number']).agg({'Number of offences': ['mean', 'min', 'max', 'sum']}))

In [42]:
year_month_df.reset_index()

Year Month number Number of offences               
                                     mean min  max   sum
0    2008            1           9.718876   1  340  2420
1    2008            2          10.155462   1  333  2417
2    2008            3          11.385281   1  392  2630
3    2008            4          10.373494   1  360  2583
4    2008            5           9.425000   1  316  2262
..    ...          ...                ...  ..  ...   ...
168  2022            1          11.501706   1  311  3370
169  2022            2          10.142857   1  248  2698
170  2022            3          11.464164   1  296  3359
171  2022            4          11.698582   1  320  3299
172  2022            5           9.884058   1  258  2728

[173 rows x 6 columns]

In [43]:
year_df = pd.DataFrame(nt_df.groupby('Year').agg({'Number of offences': ['mean', 'min', 'max', 'sum']}))

In [44]:
year_df

Number of offences                
                   mean min  max    sum
Year                                   
2008           9.991983   1  392  28667
2009           9.383984   1  347  28593
2010          10.196327   1  458  30538
2011           9.118316   1  319  27282
2012           9.679295   1  312  29638
2013           8.638737   1  304  26806
2014           8.883751   1  313  27664
2015           8.928805   1  359  27340
2016           8.994934   1  306  28406
2017           9.199754   1  290  29890
2018           9.384946   1  311  30548
2019           9.097761   1  304  28849
2020           8.893920   1  297  28087
2021           9.787698   1  346  32779
2022          10.960284   1  320  15454

In [45]:
year_df.keys()

MultiIndex([('Number of offences', 'mean'),
            ('Number of offences',  'min'),
            ('Number of offences',  'max'),
            ('Number of offences',  'sum')],
           )

In [46]:
year_df.index

Int64Index([2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018,
            2019, 2020, 2021, 2022],
           dtype='int64', name='Year')

In [47]:
year_df[year_df.index == 2008]

Number of offences                
                   mean min  max    sum
Year                                   
2008           9.991983   1  392  28667

<p style='color: red'> The number of offences is the number of offences recorded per region per statistical area. The highest number of offences ever recorded is 458 property damage offences in Darwin in January 2010. </p>

* group together reporting region and statistical area.
* statistical area with maximum offences
* look into the type of offence prevalent in statistical area.
* look into the maximum offences per region.

In [48]:
nt_df[nt_df['Number of offences'] == 458]

As At  Year  Month number          Offence category  \
6111  1/07/2022  2010             1  Property Damage Offences   

                  Offence type Alcohol involvement DV involvement  \
6111  Property damage offences                   -              -   

     Reporting region Statistical Area 2  Number of offences  
6111           Darwin                  -                 458

In [49]:
nt_df[(nt_df['Year'] == 2010) & (nt_df['Month number'] == 1) & 
      (nt_df['Offence category'] == 'Property Damage Offences')]

As At  Year  Month number          Offence category  \
6110  1/07/2022  2010             1  Property Damage Offences   
6111  1/07/2022  2010             1  Property Damage Offences   
6112  1/07/2022  2010             1  Property Damage Offences   
6113  1/07/2022  2010             1  Property Damage Offences   
6114  1/07/2022  2010             1  Property Damage Offences   
6115  1/07/2022  2010             1  Property Damage Offences   
6116  1/07/2022  2010             1  Property Damage Offences   
6117  1/07/2022  2010             1  Property Damage Offences   
6118  1/07/2022  2010             1  Property Damage Offences   
6119  1/07/2022  2010             1  Property Damage Offences   
6120  1/07/2022  2010             1  Property Damage Offences   
6121  1/07/2022  2010             1  Property Damage Offences   
6122  1/07/2022  2010             1  Property Damage Offences   
6123  1/07/2022  2010             1  Property Damage Offences   
6124  1/07/2022  2010             1  Property Damage Offences   
6125  1/07/2022  2010             1  Property Damage Offences   
6126  1/07/2022  2010             1  Property Damage Offences   
6127  1/07/2022  2010             1  Property Damage Offences   
6128  1/07/2022  2010             1  Property Damage Offences   
6129  1/07/2022  2010             1  Property Damage Offences   
6130  1/07/2022  2010             1  Property Damage Offences   
6131  1/07/2022  2010             1  Property Damage Offences   
6132  1/07/2022  2010             1  Property Damage Offences   
6133  1/07/2022  2010             1  Property Damage Offences   
6134  1/07/2022  2010             1  Property Damage Offences   

                  Offence type Alcohol involvement DV involvement  \
6110  Property damage offences                   -              -   
6111  Property damage offences                   -              -   
6112  Property damage offences                   -              -   
6113  Property damage offences                   -              -   
6114  Property damage offences                   -              -   
6115  Property damage offences                   -              -   
6116  Property damage offences                   -              -   
6117  Property damage offences                   -              -   
6118  Property damage offences                   -              -   
6119  Property damage offences                   -              -   
6120  Property damage offences                   -              -   
6121  Property damage offences                   -              -   
6122  Property damage offences                   -              -   
6123  Property damage offences                   -              -   
6124  Property damage offences                   -              -   
6125  Property damage offences                   -              -   
6126  Property damage offences                   -              -   
6127  Property damage offences                   -              -   
6128  Property damage offences                   -              -   
6129  Property damage offences                   -              -   
6130  Property damage offences                   -              -   
6131  Property damage offences                   -              -   
6132  Property damage offences                   -              -   
6133  Property damage offences                   -              -   
6134  Property damage offences                   -              -   

     Reporting region    Statistical Area 2  Number of offences  
6110    Alice Springs                     -                 200  
6111           Darwin                     -                 458  
6112        Katherine                     -                  22  
6113        Nhulunbuy                     -                   6  
6114       NT Balance             Alligator                  13  
6115       NT Balance          Anindilyakwa                   3  
6116       NT Balance                Barkly                   

In [50]:
nt_df.shape

(44994, 10)

In [51]:
nt_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 44994 entries, 0 to 44993
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   As At                44994 non-null  object
 1   Year                 44994 non-null  int64 
 2   Month number         44994 non-null  int64 
 3   Offence category     44994 non-null  object
 4   Offence type         44994 non-null  object
 5   Alcohol involvement  44994 non-null  object
 6   DV involvement       44994 non-null  object
 7   Reporting region     44994 non-null  object
 8   Statistical Area 2   44994 non-null  object
 9   Number of offences   44994 non-null  int64 
dtypes: int64(3), object(7)
memory usage: 3.4+ MB


In [52]:
nt_df = nt_df.drop('As At', axis = 1)

In [53]:
nt_df.keys()

Index(['Year', 'Month number', 'Offence category', 'Offence type',
       'Alcohol involvement', 'DV involvement', 'Reporting region',
       'Statistical Area 2', 'Number of offences'],
      dtype='object')

<h1 style="color:blue">Offence Category Counts</h1>

<span style="color:blue"> Although it is a relief that homicide and related offences have the least amount of occurences, it is quite alarming that acts intended to cause injury has the highest count among the crimes reported.</span> 

In [54]:
nt_df.groupby('Offence category')['Number of offences'].sum()

Offence category
Abduction - harassment and other offences against the person      5027
Acts intended to cause injury                                   102739
Commercial break-ins                                             29600
Homicide and related Offences                                      248
House break-ins                                                  29496
Motor theft and related offences                                 29486
Other Dangerous or negligent acts endangering persons             1787
Property Damage Offences                                        108292
Robbery - extortion and related offences                          1809
Sexual Assault and related offences                               6248
Theft and related offences (other than MV)                      105809
Name: Number of offences, dtype: int64

<p style='color: red'> There were 28,878 cases out of the total cases that were left blank. It can be assumed here that the alcohol involvement was unknown. Replace all - with Alcohol involvement unknown</p>

In [64]:
nt_df['Alcohol involvement'].unique()

array(['-', 'Alcohol involved', 'Alcohol involvement unknown',
       'No alcohol involved'], dtype=object)

In [73]:
nt_df['Alcohol involvement'] = ['Alcohol involvement unknown' if '-' in s else s for s in nt_df['Alcohol involvement']]

In [79]:
alcohol_involvement = pd.DataFrame(nt_df.groupby(['Alcohol involvement', 'Offence category'])['Number of offences'].sum()).reset_index()

In [80]:
alcohol_involvement

Alcohol involvement  \
0              Alcohol involved   
1   Alcohol involvement unknown   
2   Alcohol involvement unknown   
3   Alcohol involvement unknown   
4   Alcohol involvement unknown   
5   Alcohol involvement unknown   
6   Alcohol involvement unknown   
7   Alcohol involvement unknown   
8   Alcohol involvement unknown   
9   Alcohol involvement unknown   
10  Alcohol involvement unknown   
11  Alcohol involvement unknown   
12          No alcohol involved   

                                     Offence category  Number of offences  
0                       Acts intended to cause injury               56349  
1   Abduction - harassment and other offences agai...                5027  
2                       Acts intended to cause injury               20013  
3                                Commercial break-ins               29600  
4                       Homicide and related Offences                 248  
5                                     House break-ins               29496  
6                    Motor theft and related offences               29486  
7   Other Dangerous or negligent acts endangering ...                1787  
8                            Property Damage Offences              108292  
9            Robbery - extortion and related offences                1809  
10                Sexual Assault and related offences                6248  
11         Theft and related offences (other than MV)              105809  
12                      Acts intended to cause injury               26377

In [81]:
px.bar(alcohol_involvement, x = 'Offence category', y = 'Number of offences', color = 'Alcohol involvement')

In [84]:
dv_involvement = pd.DataFrame(nt_df.groupby(['DV involvement', 'Offence category'])['Number of offences'].sum()).reset_index()

In [88]:
dv_involvement['DV involvement'] = ['DV involvement unknown' if '-' in s else s for s in dv_involvement['DV involvement']]

In [89]:
dv_involvement

DV involvement                                   Offence category  \
0   DV involvement unknown  Abduction - harassment and other offences agai...   
1   DV involvement unknown                      Acts intended to cause injury   
2   DV involvement unknown                               Commercial break-ins   
3   DV involvement unknown                      Homicide and related Offences   
4   DV involvement unknown                                    House break-ins   
5   DV involvement unknown                   Motor theft and related offences   
6   DV involvement unknown  Other Dangerous or negligent acts endangering ...   
7   DV involvement unknown                           Property Damage Offences   
8   DV involvement unknown           Robbery - extortion and related offences   
9   DV involvement unknown                Sexual Assault and related offences   
10  DV involvement unknown         Theft and related offences (other than MV)   
11                      DV                      Acts intended to cause injury   
12  DV involvement unknown                      Acts intended to cause injury   

    Number of offences  
0                 5027  
1                  371  
2                29600  
3                  248  
4                29496  
5                29486  
6                 1787  
7               108292  
8                 1809  
9                 6248  
10              105809  
11               59546  
12               42822

In [90]:
px.bar(dv_involvement, x = 'Offence category', y = 'Number of offences', color = 'DV involvement')

In [103]:
dv_alcohol = pd.DataFrame(nt_df.groupby(['DV involvement', 'Alcohol involvement', 'Offence category'])['Number of offences'].sum()).reset_index()

In [104]:
dv_alcohol['DV involvement'] = ['DV involvement unknown' if '-' in s else s for s in dv_alcohol['DV involvement']]

In [105]:
dv_alcohol

DV involvement          Alcohol involvement  \
0   DV involvement unknown  Alcohol involvement unknown   
1   DV involvement unknown  Alcohol involvement unknown   
2   DV involvement unknown  Alcohol involvement unknown   
3   DV involvement unknown  Alcohol involvement unknown   
4   DV involvement unknown  Alcohol involvement unknown   
5   DV involvement unknown  Alcohol involvement unknown   
6   DV involvement unknown  Alcohol involvement unknown   
7   DV involvement unknown  Alcohol involvement unknown   
8   DV involvement unknown  Alcohol involvement unknown   
9   DV involvement unknown  Alcohol involvement unknown   
10  DV involvement unknown  Alcohol involvement unknown   
11                      DV             Alcohol involved   
12                      DV  Alcohol involvement unknown   
13                      DV          No alcohol involved   
14  DV involvement unknown             Alcohol involved   
15  DV involvement unknown  Alcohol involvement unknown   
16  DV involvement unknown          No alcohol involved   

                                     Offence category  Number of offences  
0   Abduction - harassment and other offences agai...                5027  
1                       Acts intended to cause injury                 371  
2                                Commercial break-ins               29600  
3                       Homicide and related Offences                 248  
4                                     House break-ins               29496  
5                    Motor theft and related offences               29486  
6   Other Dangerous or negligent acts endangering ...                1787  
7                            Property Damage Offences              108292  
8            Robbery - extortion and related offences                1809  
9                 Sexual Assault and related offences                6248  
10         Theft and related offences (other than MV)              105809  
11                      Acts intended to cause injury               36155  
12                      Acts intended to cause injury                7691  
13                      Acts intended to cause injury               15700  
14                      Acts intended to cause injury               20194  
15                      Acts intended to cause injury               11951  
16                      Acts intended to cause injury               10677

In [110]:
dv_alcohol["alcohol_dv"] = dv_alcohol[["DV involvement", "Alcohol involvement"]].apply(" & ".join, axis=1)

In [112]:
dv_alcohol.drop(['DV involvement', 'Alcohol involvement'], axis = 1)

Offence category  Number of offences  \
0   Abduction - harassment and other offences agai...                5027   
1                       Acts intended to cause injury                 371   
2                                Commercial break-ins               29600   
3                       Homicide and related Offences                 248   
4                                     House break-ins               29496   
5                    Motor theft and related offences               29486   
6   Other Dangerous or negligent acts endangering ...                1787   
7                            Property Damage Offences              108292   
8            Robbery - extortion and related offences                1809   
9                 Sexual Assault and related offences                6248   
10         Theft and related offences (other than MV)              105809   
11                      Acts intended to cause injury               36155   
12                      Acts intended to cause injury                7691   
13                      Acts intended to cause injury               15700   
14                      Acts intended to cause injury               20194   
15                      Acts intended to cause injury               11951   
16                      Acts intended to cause injury               10677   

                                           alcohol_dv  
0   DV involvement unknown & Alcohol involvement u...  
1   DV involvement unknown & Alcohol involvement u...  
2   DV involvement unknown & Alcohol involvement u...  
3   DV involvement unknown & Alcohol involvement u...  
4   DV involvement unknown & Alcohol involvement u...  
5   DV involvement unknown & Alcohol involvement u...  
6   DV involvement unknown & Alcohol involvement u...  
7   DV involvement unknown & Alcohol involvement u...  
8   DV involvement unknown & Alcohol involvement u...  
9   DV involvement unknown & Alcohol involvement u...  
10  DV involvement unknown & Alcohol involvement u...  
11                              DV & Alcohol involved  
12                   DV & Alcohol involvement unknown  
13                           DV & No alcohol involved  
14          DV involvement unknown & Alcohol involved  
15  DV involvement unknown & Alcohol involvement u...  
16       DV involvement unknown & No alcohol involved

In [113]:
px.bar(dv_alcohol, x = 'Offence category', y = 'Number of offences', color = 'alcohol_dv')

In [ ]:
nt_df.groupby('DV involvement')['Number of offences'].sum()